File: 01-string-and-math/195686/gukin-han.java
import java.io.*;
import java.util.*;
class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int numOfIngredient = Integer.parseInt(br.readLine());
		int[] ingredients = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
		boolean isGoingUp = true;
		int prev = 0;
		int answer = 0;
		
		for (int ingredient: ingredients) {
			if (isGoingUp) {
				if (ingredient >= prev) {
					answer += ingredient;
					prev = ingredient;
					continue;
				} else {
					isGoingUp = false;
				}
			}
			
			if (!isGoingUp) {
				if (ingredient <= prev) {
					answer += ingredient;
					prev = ingredient;
					continue;
				} else {
					answer = 0;
					break;
				}
			}
		}
		System.out.println(answer);
	}
}

File: 01-string-and-math/195686/gukin-han.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let count = 0;
let ingredients = [];
let isGoingUp = true;
let prev = 0;
let answer = 0;


rl.on('line', function (line) {
  count++;
  if (count === 2) {
    ingredients = line.split(' ').map(Number);
	}
}).on('close', function () {
 
	// solution
	for (let ingredient of ingredients) {
		if (isGoingUp) {
			if (ingredient >= prev) {
				prev = ingredient;
				answer += ingredient;
				continue;
			} else isGoingUp = false;
		}
		
		if (!isGoingUp) {
			if (ingredient <= prev) {
				prev = ingredient;
				answer += ingredient;
			} else {
				answer = 0;
				break;
			}
		}
	}
	console.log(answer);

  process.exit();
});
